demos/filebench/workloads/oltp.f

```diff
@@ -0,0 +1,86 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+
+# Workload "oltp" is not fully-support on Occlum.
+# Semaphore-related flowop should be commented before run.
+set $dir=/tmp
+set $eventrate=0
+set $iosize=2k
+set $nshadows=2
+set $ndbwriters=100
+set $usermode=200000
+set $filesize=800m
+set $memperthread=1m
+set $workingset=0
+set $logfilesize=10m
+set $nfiles=10
+set $nlogfiles=1
+set $directio=0
+eventgen rate = $eventrate
+
+# Define a datafile and logfile
+define fileset name=datafiles,path=$dir,size=$filesize,entries=$nfiles,dirwidth=1024,prealloc=100,reuse
+define fileset name=logfile,path=$dir,size=$logfilesize,entries=$nlogfiles,dirwidth=1024,prealloc=100,reuse
+
+define process name=lgwr,instances=1
+{
+  thread name=lgwr,memsize=$memperthread
+  {
+    flowop aiowrite name=lg-write,filesetname=logfile,
+        iosize=256k,random,directio=$directio,dsync
+    flowop aiowait name=lg-aiowait
+    flowop semblock name=lg-block,value=3200,highwater=1000
```

Comment:
Should we use aio when Occlum does not support it? I think aio is a Linux feature, not a user-space library feature. Any other features that Occlum does not support and should be worked around?

Reply:
It seems that non-supported features all appear in "oltp", which are `aio` and `semaphore` related flowops.

Reply:
Should we modify `oltp` to adapt Occlum? For example we use `randomwrite+fsync` rather than `aiowrite+aiowait`.

Reply:
I am not entirely sure. You are in a better position to make a sensible decision than I do :)

Reply:
I've already replace `aiowrite` to `write` with same parameters. `oltp` works fine now.